pmlc/conversion/gpu/gpu_to_vulkan.cc

```diff
@@ -314,16 +448,16 @@ LogicalResult ConvertGpuLaunchFuncToVulkanCalls::transferBuffers(
 
 LogicalResult ConvertGpuLaunchFuncToVulkanCalls::printBuffer(Location loc,
                                                              OpBuilder &builder,
-                                                             Value &buffer) {
+                                                             Value buffer) {
   auto type = buffer.getType();
   if (auto memRefType = type.dyn_cast_or_null<MemRefType>()) {
     auto elementType = memRefType.getElementType();
     if (elementType.isF32()) {
-      auto unrankedBuffer = builder.create<mlir::MemRefCastOp>(
+      auto unrankedBuffer = builder.create<MemRefCastOp>(
```

Comment:
We should avoid using `MemRefCastOp`, the internal lowering of a MemRefCast is to use `alloca` memory (when lowered to LLVM). If this happens inside of a loop, it's easy for the stack to be exhausted.

Reply:
I see, should we add a custom conversion pattern for MemRefCastOp to llvm and avoid use `alloca` memory?

Reply:
I see, this is a very good point, I will think more about it.